### ISSUE TITLE
Record when a file only contains partial data and overwrite on next run

### ISSUE THREAD

**Issue by semiligneous** [zigam/tesla-solar-download#2]
**Problem:**

Running the script will generate 1 file per day starting today. Files for "yesterday" and further back in time will have full data (for every 5-minte interval), but the file created for "today" will be missing data between "now" and 23:55pm. The rest of the file will be filled with zeros, eg:

```
tail -n 10 2023-06-04.csv
2023-06-04 23:10:00,0,0,0,0,0,0
2023-06-04 23:15:00,0,0,0,0,0,0
2023-06-04 23:20:00,0,0,0,0,0,0
2023-06-04 23:25:00,0,0,0,0,0,0
2023-06-04 23:30:00,0,0,0,0,0,0
2023-06-04 23:35:00,0,0,0,0,0,0
2023-06-04 23:40:00,0,0,0,0,0,0
2023-06-04 23:45:00,0,0,0,0,0,0
2023-06-04 23:50:00,0,0,0,0,0,0
2023-06-04 23:55:00,0,0,0,0,0,0
```

When you run the script again "tomorrow", or any future time, it will pull data only for days that do not have a file which already exists - in this example it would create a new (partially complete) file for 2023-06-05, and then stop, because 2023-06-04.csv already exists.

However, now I have an incomplete record of data, with missing data points in the late stages of the day I ran the script (2023-05-04 in this example).

**Current workaround:**

Manually delete or rename the partially complete file, so that upon next / future run, a new complete file will be created. 

**Proposed / possible solution:**

1. if a file does not have complete data, which it never will for "today's" date (unless run at 23:59:59), name the file 2023-06-05_partial.csv
2. When the script runs next time, ignore any _partial.csv files, which would result in a new (complete) 2023-06-05.csv file getting created. 
3. Delete the _partial.csv file so that there are no duplicate data points in the directory

Thanks

**Comment by zigam** [comment]
Thanks for the detailed bug report, fixed in bdb96458590fba8d52511af419b9bac5ec6a7d41.

**Comment by semiligneous** [comment]
thanks @zigam works perfectly 👍
